app/Http/Requests/UpdateManagerProfileRequest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use App\Services\Validation\Rules\ValidIdRule;
+use App\Models\Lookup\Frequency;
+use App\Models\Lookup\Department;
+use App\Services\Validation\Rules\LinkedInUrlRule;
+use Illuminate\Validation\Rule;
+use App\Services\Validation\Rules\PasswordCorrectRule;
+use App\Services\Validation\Rules\PasswordFormatRule;
+use App\Services\Validation\Rules\TwitterHandleRule;
+
+class UpdateManagerProfileRequest extends FormRequest
+{
+    /**
+     * Validator instance updated on failedValidation.
+     *
+     * @var \Illuminate\Contracts\Validation\Validator
+     */
+    public $validator = null;
+
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return boolean
+     */
+    public function authorize()
+    {
+        $manager = $this->route('manager');
+        return $manager->id !== null;
+    }
+
+    /**
+     * Override Handle a failed validation attempt.
+     *
+     * @return void
+     */
+    protected function failedValidation(\Illuminate\Contracts\Validation\Validator $validator)
+    {
+        $this->validator = $validator;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $frequencyRule = new ValidIdRule(Frequency::class);
+        return [
+            'name' => 'required|string|max:191',
+            'email' => [
+                'required',
+                'string',
+                'max:191',
+                'email',
+                // Email may match existing email for this user,
+                // but must be unique if changed.
+                Rule::unique('users', 'email')->ignore($this->manager->user->id)
```

Comment:
Are you sure `$this->manager` will work in this class?

Reply:
Seems to be working.